app/src/main/java/org/redcross/openmapkit/tagswipe/TagSwipeActivity.java

```diff
@@ -65,6 +100,225 @@ protected void onCreate(Bundle savedInstanceState) {
+                    , true, false, false, false, true, true, true,
+                    Criteria.POWER_LOW, Criteria.ACCURACY_FINE);
+            changeTestProviderEnabled(true);
+            locationManager.requestLocationUpdates(preferredLocationProvider, 0, 0, locationListener);
+        }
+
+        updateUsersLocation();
+    }
+
+    /**
+     * This method changes the state of the test provider initialized in this activity.
+     * This method is intended to only be used in tests.
+     *
+     * @param enable    TRUE of you want to enable the test provider
+     */
+    public void changeTestProviderEnabled(boolean enable) {
+        if(locationManager != null) {
+            locationManager.setTestProviderEnabled(preferredLocationProvider, enable);
+        } else {
+            Log.w("TagSwipeActivity", "Location manager is null, cannot enable/disable the test provider");
+        }
+    }
+
+    /**
+     * This method changes the status of the test provider for the locationManager initialized in
+     * this activity.
+     * This method is intended to only be used in tests.
+     *
+     * @param status    Status similar to those provided in android.location.LocationProvider
+     * @see android.location.LocationProvider
+     */
+    public void changeTestProviderStatus(int status) {
+        if(locationManager != null) {
+            locationManager.setTestProviderStatus(
+                    preferredLocationProvider,
+                    status,
+                    null,
+                    Calendar.getInstance().getTimeInMillis());
+        } else {
+            Log.w("TagSwipeActivity", "Location manager is null, cannot change the status of the test provider");
+        }
+    }
+
+    /**
+     * This method changes the location passed by the test provider registered in the
+     * locationManager initialized in this activity.
+     * This method is intended to only be used in tests.
+     *
+     * @param location  The location to be provided by the locationManager
+     */
+    public void changeTestProviderLocation(Location location) {
+        if(locationManager != null) {
+            locationManager.setTestProviderLocation(preferredLocationProvider, location);
+        } else {
+            Log.w("TagSwipeActivity", "Location manager is null, cannot change the location in the test provider");
+        }
+    }
+
+    public String getPreferredLocationProvider() {
+        return preferredLocationProvider;
+    }
+
+    /**
+     * This method updates the user's location (and osm location tags)
+     *
+     * @return TRUE if location was successfully updated
+     */
+    public boolean updateUsersLocation() {
+        if(checkLocationProviderEnabled()) {
+            Log.w("GPSTest", "LocationManager is not null");
+            Location location = locationManager.getLastKnownLocation(preferredLocationProvider);
+            if(location == null) {
+                Log.w("GPSTest", "Last location is null");
```

Comment:
This test logs, were they to be removed or they should still be present? If present, should the logging use the class name and class method be part of the log or is this automatically added?